cloudwatch/cloudwatch.go

```diff
@@ -584,6 +584,20 @@ func (output *OutputPlugin) putLogEvents(stream *logStream) error {
 				stream.nextSequenceToken = &parts[len(parts)-1]
 
 				return output.putLogEvents(stream)
+			} else if awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+				// a log group or a log stream should be re-created after it is deleted
+				logrus.Infof("[cloudwatch %d] Encountered error %v; detailed information: %s\n", output.PluginInstanceID, awsErr, awsErr.Message())
+				if strings.Contains(awsErr.Message(), "group") {
+					if err := output.createLogGroup(&Event{group: stream.logGroupName}); err != nil {
+						logrus.Error(err)
+						return err
+					}
+				} else if strings.Contains(awsErr.Message(), "stream") {
+					if _, err := output.createStream(&Event{group: stream.logGroupName, stream: stream.logStreamName}); err != nil {
+						logrus.Error(err)
+						return err
+					}
+				}
 			} else {
```

Comment:
The only problem with this code is that after this error, you should return and let it retry. If you continue the code after this, I think that should cause a problem since the PutLogEvents request failed and the rest of the code assumes it succeeded.

So you need to still return an error even if you succeeded in re-creating the log group or stream. 

I recommend something like:
```
return fmt.Errorf("Log group did not exist, re-created it. Will retry PutLogEvents on next flush");
```

Reply:
Thanks for the suggestion. Will add a return statement here..